torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -876,7 +902,16 @@ TensorView* TensorViewBuilder::build() const {
           shape_[i] >= 0,
           "Invalid extent value. ",
           "For a tensor representing a single scalar use ndims = 0 with no sizes set.");
-      domain[i] = new IterDomain(new Int(0), new Int(shape_[i]));
+      if (shape_[i] == 1) {
+        // If size is known to be 1, assume it needs to be broadcasted.
+        domain[i] = new IterDomain(
+            new Int(0),
+            new Int(1),
+            ParallelType::Serial,
+            IterType::BroadcastWithStride);
+      } else {
```

Comment:
This makes sense, but how is it related to `view`?

Reply:
On the python side, any size-1 dimension is promoted to be a broadcast dimension. This change made it easier to replicate this behavior for testing and debugging.